src/FloatingWindows/ImGUIIntegration.cpp

```diff
@@ -54,6 +54,17 @@ ImGUIWindow::ImGUIWindow(int width, int height, int decoration):
     io.KeyMap[ImGuiKey_Y] = XPLM_VK_Y;
     io.KeyMap[ImGuiKey_Z] = XPLM_VK_Z;
 
+    font0 = io.Fonts->AddFontDefault();
+    font1 = io.Fonts->AddFontFromFileTTF("./Resources/fonts/DejaVuSans.ttf", 13.0f);
```

Comment:
You are loading all the fonts every time you create a window, which is probably unnecessary. I think `initFloatingWindowSupport()` is a better place for it.

And to make PsuhFont() work, you need to expose font1, ..., font9 as lua variables like so:

```cpp
::sol::state_view lua(L);
lua["FONT1"] = io.Fonts->AddFontFromFileTTF("./Resources/fonts/DejaVuSans.ttf", 13.0f);
//...
```

Didn't test this, but I think it should work with no or little tweaking :)

P.S. I'd vote for non-numeric font names, e.g. FONT_ROBOTO_LIGHT.